### DISCUSSION TITLE
Running without remote debugging

### DISCUSSION THREAD

**Discussion by cjhines** [microsoft/react-native-windows#10411] [Q&A] [answered]
I'm noticing a few performance issues while the remote debugging is connected, and wanted to run without it.

Hot-reloading the app from the developer menu often crashes the app, and the following command still opens the debugger:

`npx react-native run-windows --remote-debugging=false`

Any advice?

**Update:** Clicking 'Disable remote debugging' also freezes the app.

**Comment by chrisglein** [comment]
Lots of issues you're mentioning here.

> Hot-reloading the app from the developer menu often crashes the app

Well that's no good. Do you have a call stack? Or is this overlapping with you not having the remote debugger?

> Clicking 'Disable remote debugging' also freezes the app.

I know this is happening as you're trying to disable debugging, but have you put the app in the debugger to know why it's frozen?

But yes, remote debugging is slow :(

**Comment by mlecoq** [comment]
why `npx react-native run-windows --remote-debugging=false` does not disable remote debugging at startup ?

**Reply by chrisglein** [reply]
This thread is over a year old now. It'd be helpful to know what version you're running here (`npx react-native info`) and whether you have any changes made to your app to use Hermes or leave the default Chakra JavaScript engine.

**Comment by acoates-ms** [comment, marked as answer]
The --remote-debugging flag works by passing a command line arg to the application.  Its up to the application to handle that argument.

The default implementation of `ReactApplication` handles the argument here:
https://github.com/microsoft/react-native-windows/blob/f859b4ccfa07c173dfb1b4a48286342b0d56dbdf/vnext/Microsoft.ReactNative/ReactApplication.cpp#L153C20-L153C20

If you are handing activation differently, or not using `ReactApplication`, then you would have to do your own command line argument handling.  You'll also see in the linked code, that the argument is only handled in DEBUG builds.  If you just want to disable remote debugging, you can do that in your own code by settting `InstanceSettings().UseWebDebugger(false);`
